weblate/addons/git.py

```diff
@@ -60,6 +60,34 @@ def get_filenames(self, component):
                 languages[code].extend(translation.filenames)
         return languages
 
+    def get_squash_commit_message(self, repository, log_format, remote, filenames=None):
+        commit_message = self.instance.configuration.get("commit_message")
+
+        if not commit_message:
+            command = [
+                "log",
+                "--format={}".format(log_format),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            commit_message = repository.execute(command)
+
+        if self.instance.configuration.get("append_trailers"):
+            command = [
+                "log",
+                "--format={}".format("%(trailers:separator)"),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            trailers = repository.execute(command)
+            commit_message = "\n\n".join([commit_message, trailers])
```

Comment:
Won't this include trailers twice in case custom commit message is not set? First time they would be included from the commit message and now for the second time.

What I think would make more sense is to avoid trailers from the squashed messages and add them here separately. Though I don't see way to do this using git log.

Reply:
I also couldn't find a way to do this with `git log`. I ended up needing to remove the trailers from the commit message anways because otherwise repeated squashes would result in an exponential increase in trailers.